system/security/shared/caching.cpp

```diff
@@ -435,17 +432,16 @@ void CPermissionsCache::add(ISecUser& sec_user)
 #ifdef _DEBUG
     DBGLOG("CACHE: CPermissionsCache Adding cached user %s", username);
 #endif
-    m_userCache[username] = new CachedUser(sec_user.clone());
+    m_userCache[username] = new CachedUser(LINK(&sec_user));
```

Comment:
The only way I can see protecting against that is to revert the code back to the clone().   Which do you prefer?